Replicator/tests/ReplicatorSGTest.cc

```diff
@@ -1388,3 +1388,200 @@ TEST_CASE_METHOD(ReplicatorSGTest, "Pinned Certificate Success", "[.SyncServer]"
+    _initParams.propertyEncryptor = &testEncryptorError;
+    _encCBContext = &encryptContext;
+
+    SECTION("LiteCoreDomain, kC4ErrorCrypto") {
+        encryptContext.simulateError = C4Error {LiteCoreDomain, kC4ErrorCrypto};
+        _expectedDocPushErrors = { "seekrit" };
+        replicate(kC4OneShot, kC4Disabled);
+        CHECK(_callbackStatus.progress.documentCount == 2);
+        CHECK(encryptContext.called == 1);
+
+        // Try again with good encryptor and we get the encrypted doc.
+        _initParams.propertyEncryptor = &testEncryptor;
+        _expectedDocPushErrors = {};
+        replicate(kC4OneShot, kC4Disabled);
+        // Crypto error will mark the rev as already synced.
+        CHECK(_callbackStatus.progress.documentCount == 0);
+        CHECK(encryptContext.called == 1);
+    }
+
+    SECTION("WebSocketDomain/503") {
+        encryptContext.simulateError = C4Error {WebSocketDomain, 503};
+        _mayGoOffline = true;
+        replicate(kC4OneShot, kC4Disabled);
+        CHECK(_wentOffline);
+        CHECK(encryptContext.called == 2);
+
+        Encoder enc;
+        enc.beginDict();
+        enc.writeKey(C4STR(kC4ReplicatorOptionDisablePropertyDecryption));
+        enc.writeBool(true);
+        enc.endDict();
+        _options = AllocedDict(enc.finish());
+        deleteAndRecreateDB();
+        replicate(kC4Disabled, kC4OneShot);
+        CHECK(c4db_getDocumentCount(db) == 3);
+    }
+}
+
+TEST_CASE_METHOD(ReplicatorSGTest, "Replicate Decryptor Error", "[.SyncServer]") {
+    _remoteDBName = kScratchDBName;
+    flushScratchDatabase();
+
+    slice originalJSON = R"({"SSN":{"@type":"encryptable","value":"123-45-6789"}})"_sl;
+    slice unencryptedJSON = R"({"ans*wer": 42})"_sl;
+    {
+        TransactionHelper t(db);
+        createFleeceRev(db, "doc01"_sl, kRevID, unencryptedJSON);
+        createFleeceRev(db, "seekrit"_sl, kRevID, originalJSON);
+        createFleeceRev(db, "doc03"_sl, kRevID, unencryptedJSON);
+    }
+
+    TestEncryptorContext encryptContext = {"seekrit", "SSN"};
+    _initParams.propertyEncryptor = &testEncryptor;
+    _encCBContext = &encryptContext;
+    replicate(kC4OneShot, kC4Disabled);
+
+    // check the 3 documents are pushed and clear the local db
+    // Get ready for Pull/Decyption
+    CHECK(c4db_getDocumentCount(db) == 3);
+    deleteAndRecreateDB();
+    _encCBContext = NULL;
+    TestEncryptorContext decryptContext = {"seekrit", "SSN"};
+    _initParams.propertyDecryptor = &testDecryptorError;
+    _decCBContext = &decryptContext;
+
+    SECTION("LiteCoreDomain, kC4ErrorCrypto") {
+        decryptContext.simulateError = C4Error {LiteCoreDomain, kC4ErrorCrypto};
+        _expectedDocPullErrors = { "seekrit" };
+        replicate(kC4Disabled, kC4OneShot);
+        CHECK(_callbackStatus.progress.documentCount == 2);
+        CHECK(decryptContext.called == 1);
+
+        // Try again with good decryptor and we get the encrypted doc.
```

Comment:
Not sure what "we get encrypted doc" means in this context.

Reply:
copy/paste mistake.